src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderProgramProfile.kt

```diff
@@ -0,0 +1,130 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa
+
+import jakarta.persistence.Column
+import jakarta.persistence.Entity
+import jakarta.persistence.FetchType
+import jakarta.persistence.GeneratedValue
+import jakarta.persistence.Id
+import jakarta.persistence.JoinColumn
+import jakarta.persistence.ManyToOne
+import jakarta.persistence.SequenceGenerator
+import jakarta.persistence.Table
+import org.hibernate.Hibernate
+import java.time.LocalDate
+
+@Entity
+@Table(name = "OFFENDER_PROGRAM_PROFILES")
+data class OffenderProgramProfile(
+  @Id
+  @Column(name = "OFF_PRGREF_ID")
+  @SequenceGenerator(name = "OFF_PRGREF_ID", sequenceName = "OFF_PRGREF_ID", allocationSize = 1)
+  @GeneratedValue(generator = "OFF_PRGREF_ID")
+  val offenderProgramReferenceId: Long = 0,
+
+  @ManyToOne(optional = false, fetch = FetchType.LAZY)
+  @JoinColumn(name = "OFFENDER_BOOK_ID", nullable = false)
+  val offenderBooking: OffenderBooking,
+
+  @ManyToOne(optional = false, fetch = FetchType.LAZY)
+  @JoinColumn(name = "PROGRAM_ID", nullable = false)
+  val program: ProgramService,
+
+  @Column(name = "OFFENDER_START_DATE")
+  val startDate: LocalDate? = null,
```

Comment:
again is this really nullable?

Reply:
There seems to be plenty of rows in prod which have start date or agency null, so best to leave as is